cylc/flow/scheduler.py

```diff
@@ -561,17 +490,159 @@ def configure(self):
+        )
+        LOG.info(
+            'Run: (re)start=%d log=%d', n_restart, 1, extra=log_extra_num)
+        LOG.info('Cylc version: %s', CYLC_VERSION, extra=log_extra)
+        # Note that the following lines must be present at the top of
+        # the suite log file for use in reference test runs:
+        LOG.info('Run mode: %s', self.config.run_mode(), extra=log_extra)
+        LOG.info(
+            'Initial point: %s', self.config.initial_point, extra=log_extra)
+        if self.config.start_point != self.config.initial_point:
+            LOG.info(
+                'Start point: %s', self.config.start_point, extra=log_extra)
+        LOG.info('Final point: %s', self.config.final_point, extra=log_extra)
+
+    async def start_scheduler(self):
+        """Start the scheduler main loop."""
+        try:
+            self.data_store_mgr.initiate_data_model()
+            self._configure_contact()
+            self.run_event_handlers(self.EVENT_STARTUP, 'suite starting')
+            await asyncio.gather(
                 *main_loop.get_runners(
                     self.main_loop_plugins,
                     main_loop.CoroTypes.StartUp,
                     self
                 )
             )
-        )
+            await self.publisher.publish(
+                self.data_store_mgr.get_publish_deltas()
+            )
+            self.profiler.start()
+            await self.main_loop()
 
-        self.profiler.log_memory("scheduler.py: end configure")
+        except SchedulerStop as exc:
+            # deliberate stop
+            await self.shutdown(exc)
+            if self.auto_restart_mode == AutoRestartMode.RESTART_NORMAL:
+                self.suite_auto_restart()
+            # run shutdown coros
+            await asyncio.gather(
+                *main_loop.get_runners(
+                    self.main_loop_plugins,
+                    main_loop.CoroTypes.ShutDown,
+                    self
+                )
+            )
+
+        except SchedulerError as exc:
+            await self.shutdown(exc)
+            raise exc from None
+
+        except Exception as exc:
+            try:
+                await self.shutdown(exc)
+            except Exception as exc2:
+                # In case of exceptions in the shutdown method itself
+                LOG.exception(exc2)
+            raise exc from None
+
+        else:
+            # main loop ends (not used?)
+            await self.shutdown(SchedulerStop(StopMode.AUTO.value))
+
+        finally:
+            self.profiler.stop()
+
+    async def run(self, daemonize=False):
+        """Run the startup sequence.
+
+        * initialise
+        * configure
+        * start_servers
+        * start_scheduler
+
+        Lightweight wrapper for convenience.
+
+        """
+        try:
+            await self.install()
+            await self.initialise()
+            await self.configure()
+            await self.start_servers()
+            await self.log_start()
```

Comment:
> so I guess you did this in anticipation of better use of asyncio in the future?

Actually some of them do await things, so I thought it would be better to use an await interface for all of them to avoid having to individually upgrade them later.